lib/registerPlugins.js

```diff
@@ -18,4 +24,45 @@ module.exports = (server, callback) => {
             }
         }, next);
     }, callback);
+}
+
+/**
+ * User Plugin. A plugin that the user wishes for the Screwdriver API to register and use
+ * @typedef {Object} UserPlugin
+ * @property {Object} register  A HapiJS plugin `register` function. It honors the interface
+ *                              that all HapiJS plugins implement.
+ * @property {Object} options  Additional options to load with the plugin.
+ */
+
+/**
+ * Register any user-requested plugins
+ * @method registerAdditionalPlugins
+ * @param  {Object}                 server    HapiJS server object
+ * @param  {Array.<UserPlugin>}     plugins   An array of plugins to register
+ * @param  {Function}               callback  Function to invoke when completed
+ */
+function registerAdditionalPlugins(server, plugins, callback) {
```

Comment:
Why are we adding plugins?  Isn't this supposed to just be adding config like what datastore system to use?